### PR TITLE
Improve-genericness

### PR DESCRIPTION
**Pull request by kvantricht**
Two small updates:

- rename the weights loading method to avoid confusion between loading a pretrained model and loading potentially an entire model including head. `pretrained_model_path` should not be used when loading a finetuned model, as its head will be overwritten. For downstream inference, we should always use `load_presto_weights` after fully constructing the Presto model.
- We should not raise an error when a head is present and no labels are provided to `Predictors` to be able to run a forward pass during inference when a head is included.

### CODE REVIEW

**Review by gabrieltseng — APPROVED** [review]
LGTM!
